### ISSUE TITLE
Volumes are not declared, and are not removed

### ISSUE THREAD

**Issue by zephod** [aws/amazon-ecs-cli#442]
### Summary

My ECS Container Instances are quickly eating up all their HDD by filling up `/var/lib/docker/volumes`. Once full, they can no longer spin up/down images and cannot even process commands to remove exited containers.

I think this is because of the way ECS CLI declares `volumes_from`. Otherwise it is a bug in ECS Agent.

### Description

I am using two Docker images in one service:

* One derived from [an official WordPress Dockerfile](https://github.com/docker-library/wordpress/blob/59f3b513af128d12da1403721e4f9be8d882ec54/php7.2/fpm-alpine/Dockerfile)
* One derived from [an official Nginx Dockerfile](https://github.com/nginxinc/docker-nginx/blob/dbd053d52727bc8db0fec704caa22b8e0d5f6c84/mainline/stretch/Dockerfile)

My compose file is as follows:

    version: "2"
    services: 
      wordpress:
        image: $REPO/wordpress:commit-$COMMIT
      nginx:
        image: $REPO/nginx:commit-$COMMIT
        ports:
         - 0:80
        environment:
          POST_MAX_SIZE: 2m
        volumes_from:
         - wordpress

The command I'm using to update the service is:

    export COMMIT=....
    /usr/local/bin/ecs-cli compose --cluster=$CLUSTER --file=$YAML_FILE --project-name=wordpress service up --deployment-min-healthy-percent=0

I'm using ECS CLI version `ecs-cli version 1.3.0 (3363f79)`.

### Expected Behavior

I'd like to replace the running version of this service. The service is still in development, so downtime is acceptable, hence the `--deployment-min-healthy-percent=0`.

I'd expect to be able to run frequent deploys, ie. several times per day.

### Observed Behavior

The WordPress instance works, and it creates a ~100mb volume shared on the host between the two running containers. This contains the whole running WordPress install and our custom code (see the `VOLUME /var/www/html` line in their Dockerfile).

However after 30-50 deploys, the Container Instance (`r4.large`) becomes unusable because there is no hard drive space remaining.

The AWS ECS Agent is:

* Not cleaning up the stopped containers,
* Not removing the previous images,
* Not removing any Volumes. (~116MB each!)

The two containers are being automatically configured as follows by ECS CLI. 

WordPress:

<img width="407" alt="screen shot 2018-03-10 at 14 24 37" src="https://user-images.githubusercontent.com/92938/37243287-eee23b44-246e-11e8-95e8-0a964f39ea3b.png">

Nginx:

<img width="415" alt="screen shot 2018-03-10 at 14 24 43" src="https://user-images.githubusercontent.com/92938/37243291-f4425ace-246e-11e8-8fed-697522830f4d.png">

I notice that the Dockerfile's VOLUME is not declared. Should this be declared explicitly in the compose file? Will this cause it to be cleaned up correctly?

I apologise if this turns out not to be a bug in ECS CLI but it is a very confusing situation. Many thanks in advance for your help.

### Config files

As above. And I'm not using an ecs-params.yml file.

**Comment by allisaurus** [comment]
Hi @zephod  - can you tell me how quickly you're deploying updates to your service? The default cleanup period after a container exits is 3 hours, so if you're deploying many times within that space it's possible disc space could run out before exited containers have been removed.

If you believe this is what's happening, it's possible to configure this value ("ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION" in the ECS agent config) to something shorter at the time your container instance spins up: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-config.html

Hopefully this helps, but please let me know if you think something else is going on or if you have follow up questions.

**Comment by zephod** [comment]
Many thanks: I've identified that this is because the container is getting stuck in a reboot loop, which seems to be a very common outcome when there is a slight misconfiguration. The difficulty has been identifying that a reboot loop is silently happening (in this case, because of a failed health check) and how to fix it.

Because of this, a new r4.large container instance would become unusable within about 45 minutes. I've solved this by fixing the health check, which in turn I fixed by adding a trailing slash to the appropriate path in config.

I think there's work to be done on this orchestration system but it's clearly not ecs-cli's fault so this issue is filed in the wrong place. I will close this ticket :-)
